Fix all entries with wrong output for number_to_the_power_of.__doc__

mlistewnik opened the pull request:
Hi @BethanyG, I read your comments to the previous PR and checked for all instances of the docstring to `number_to_the_power_of` function. 
I found one more instance, altogether 3 files were changed.

This PR relates to issue: https://github.com/exercism/python/issues/2833

Note 1: Prettier suggests to add one more line after `>>> print(number_to_the_power_of.__doc__)` but I kept the same distance as in the original, because it's supposed to show how the REPL prints output, so I guess not having that blank line is fine?

Note 2: I wanted to update the already created PR (#2834), but some of my changes (updated branch to be in sync with main) must have automatically closed it.

BethanyG commented:
@mlistewnik --  I think that between me leaving comments and when you tried to add additional changes we merged several other PRs, so that might have done something to yours -- although you should have been able to pull changes an rebase and continue.  Strange.

BethanyG reviewed (approved):
Thank you for being so through, and for finding that last example. 🌟 

We really appreciate this PR - nice work!

mlistewnik commented:
Hi,
@BethanyG I had a think about this closed PR and remembered that I corrected some empty tabs in the original commit by git reset, added the two new files and re-pushed in one commit. So that must have caused the PR to close.
And I'm new to rebasing, but I'm getting the hang of it slowly. One thing I'm wondering in the aftermath:
If I had updated the PR with a new commit, making a rebase to accommodate any changes to main branch, then would the first, earlier commit appear in the history before the new commits from merged PRs that were merged in the meantime? So, in effect, would the history look like this (newest from left):
`my_new_commit_added_to_PR (feature,origin/main) <-- commit_from_user_A < -- commit_from_user_B <-- my_initial_commit`?

BethanyG commented:
Hi @mlistewnik  - I will not even pretend to be any sort of `git` expert - for that, you'll want to have a discussion with our resident guru @kotp.  But my understanding of `git rebase` is that it rewrites history to form a "linear" commit chain.  Commits get "inserted" into the chain at the point they were merged instead of being inserted in one merge commit. 

This Atlassian article on [rebasing](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) helps explain the scenario, and this one on the [golden rule of rebasing](https://www.atlassian.com/git/tutorials/merging-vs-rebasing#the-golden-rule-of-rebasing) explains some of the pitfalls.

If you had rebased, the merged PRs would be added into your commit chain based on when they were introduced to `main`  (**_@kotp please correct me if I am wrong here!!_**), so yes -- those changes might appear _before_ your newest changes -- but after your first ones.

kotp commented:
On 22/01/03 10:56AM, BethanyG wrote:
> This Atlassian article on
> [rebasing](https://www.atlassian.com/git/tutorials/merging-vs-rebasing)
> helps explain the scenario, and this one on the [golden rule of
> rebasing](https://www.atlassian.com/git/tutorials/merging-vs-rebasing#the-golden-rule-of-rebasing)
> explains some of the pitfalls.

Good resources, (a little bit too "scary" on the golden rule part, in
that "public" is a little bit too broad but it) gives a good baseline to
work from.

> If you had rebased, the merged PRs would be added into your commit
> chain based on when they were introduced to `main`  ***@***.*** please
> correct me if I am wrong here!!_**), so yes -- those changes might
> appear _before_ your newest changes -- but after your first ones.

If the branch or your repository where the work was done had rebased
your changes onto `exercism/python/main` then that would be OK, but you
do not want to rebase main onto your changes.  But as long as we
understand the target of the rebase, and that target is (usually)
`origin/main` then yes, that is correct.

mlistewnik commented:
Thank you, @BethanyG  and @kotp  for your kind answers. The Atlassian tutorial was a great read. I particularly liked the different scenarios and visualisations of the commit tree. This PR was a delightful experience and I hope to be able to contribute in the future, too.
